pcdsdevices/lodcm.py

```diff
@@ -1137,9 +1472,124 @@ def calc_energy(self, energy, material=None, reflection=None):
         """
         # try to determine the material and reflection:
         material = material or self.get_material()
-        reflection = reflection or self.get_reflection(
-            as_tuple=True, check=True)
-        return self.energy.calc_energy(energy, material, reflection)
+        reflection = reflection or self.get_reflection()
+        return self.energy.calc_geometry(energy, material, reflection)
+
+    def set_energy(self, energy, material=None, reflection=None):
+        """
+        Set the current positions of th1, th2, z1 and z2 offset motors.
+
+        Parameters
+        ----------
+        energy : number
+            Energy in keV.
+        material : str, optional
+            Crystal material. E.g.: `Si`
+        reflection : tuple, optional
+            Crystal's reflections. E.g.: `(1,1,1)`
+        """
+        # try to determine the material and reflection:
+        material = material or self.get_material()
+        reflection = reflection or self.get_reflection()
+        th, z = diffraction.get_lom_geometry(energy * 1e3, material,
+                                             reflection)
+        # TODO: probably not the correct approach here to use
+        # set_current_position
+        if material == 'Si':
+            self.th1Si.set_current_position(th)
+            self.th2Si.set_current_position(th)
+            self.z1Si.set_current_position(-z)
+            self.z2Si.set_current_position(z)
+        elif material == 'C':
+            self.th1C.set_current_position(th)
+            self.th2C.set_current_position(th)
+            self.z1C.set_current_position(-z)
+            self.z2C.set_current_position(z)
+        else:
+            raise ValueError("Invalid material: %s", material)
+
+    def tweak_x(self, x_value, material=None, wait=False):
+        """
+        Tweak some motors to center the beam on YAG2 horizontally based on
+        the calibrated s4 position.
+
+        Parameters
+        ----------
+        x_value : number
+            The value that we want to move `x2` by from current position.
+        material : str, optional
+            Crystal material. E.g.: `Si`
+        wait : bool, optional
+            Wait for motion to complete if wait is `True`. Defaults to `False`.
+        """
+        # try to determine the material:
+        material = material or self.get_material()
+        if material == 'Si':
+            th = self.th2Si.wm()
+        elif material == 'C':
+            th = self.th2C.wm()
+        else:
+            raise ValueError("Invalid material: %s", material)
+        z_value = x_value / np.tan(2 * th)
```

Comment:
Are there issues here with angular units? `np.tan` is in radians, I thought all the motors were in degrees.

Reply:
Oh thanks for catching this, yeah I think i need to convert that to degrees first

Reply:
Fixed this in 4d32fc3 - I had my test wrong too because I manually calculated the values of what they should be - I thought it was simple math 🙅